index.bs

```diff
@@ -795,6 +795,89 @@ Sequence:
   </pre>
 </div>
 
+
+## Period datatype ## {#period-datatype}
+
+<pre class='railroad'>
+Choice:
+  Sequence:
+    N: start
+    T: /
+    N: end
+  Sequence:
+    N: start
+    T: /
+    N: duration
+  Sequence:
+    N: duration
+    T: /
+    N: end
+  T: duration
+</pre>
+
+Where "start" and "end" are [[#datetime-datatype]] and "duration" is defined
+as:
+
+<pre class='railroad'>
+T: P
+OneOrMore:
+  Stack:
+    Sequence:
+      Sequence:
+        Optional:
+          Sequence:
+            N: 0-9
+            N: 0-9
+            N: 0-9
+            N: 0-9
+            T: Y
+        Optional:
+          Sequence:
+            N: 0-9
+            T: M
+        Optional:
+          Sequence:
+            N: 0-9
+            N: 0-9
+            T: D
+    Sequence:
+      Sequence:
+        T: T
+        Optional:
+          Sequence:
+            N: 0-9
+            N: 0-9
+            T: H
+        Optional:
+          Sequence:
+            N: 0-9
+            N: 0-9
```

Comment:
I'm wondering why there are only 2 numbers allowed here - is it because you're only permitted values up to 59? In that case should the first be `N:0-5`? I'm a bit new to these diagrams so these are possibly silly questions.

Reply:
No, you are right. Also I'm using here this type of diagram to keep consistency with previous work but I'm unpleased with the visual complexity they show when patterns get too complex. In a future version I'll consider moving away from it, maybe to EBNF or similar textual expression format.

Reply:
I'll review the entire diagram. Any atom supports from 0 to n as we are expressing amounts of that atom (e.g. you can have 601 minutes `PT600M` which is equivalent to `PT10H1M`. Both are valid expressions of the same* amount of time.

(*) Assuming same accuracy.

Reply:
The new version should reflect better all valid patterns.